libvirt/tests/src/virsh_cmd/domain/virsh_domfsinfo.py

```
@@ -191,6 +191,9 @@ def hotplug_domain_disk(domain, target, source=None, hotplug=True):
             result = virsh.attach_disk(domain, source, target, "--live",
                                        ignore_status=False, debug=True)
         else:
+            session = vm.wait_for_login()
+            session.cmd("umount %s" %mount_dir)
```

Could you help give any explanation for the unmounting?

---

> Could you help give any explanation for the unmounting?
if I do not unmount the disk on the guest before I unplug the disk ,
when I run the command "virsh domfsinfo avocado-vm-1" ,
I will get an error message  as followed:
"error: Unable to get filesystem information
unable to execute QEMU agent command 'guest-get-fsinfo' "

---

Oh, yes, you need because you did mount after attaching the disk.
So could you help put the line from 194 to line 196 into a try block because they maybe fail as unexpected?

---

> Oh, yes, you need because you did mount after attaching the disk.
> So could you help put the line from 194 to line 196 into a try block because they maybe fail as unexpected?

hi I have tried to put the line from 194 to line 196 into a try block .
Do there 

> Oh, yes, you need because you did mount after attaching the disk.
> So could you help put the line from 194 to line 196 into a try block because they maybe fail as unexpected?

Hi，I have  already put the line from 194 to line 196 into a try block .
Is there anything else I need to do to improve this test code?